### PR TITLE
Kraken: update all identical cause, severity and tag

### PR DESCRIPTION
**Pull request by antoine-de**
when chaos change a cause, severity or tag we did not take the change
into account.

this fixes it

fixes http://jira.canaltp.fr/browse/NAVITIAII-2223

### CODE REVIEW

File: source/kraken/tests/fill_disruption_from_chaos_tests.cpp [hunk 1]
```diff
@@ -469,3 +469,90 @@ BOOST_AUTO_TEST_CASE(add_impact_on_line_section) {
...
+            ("stop1", "15:00"_t)
+            ("stop2", "16:00"_t);
+
+    navitia::type::Data data;
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->meta->production_date = boost::gregorian::date_period(boost::gregorian::date(2012,6,14), boost::gregorian::days(7));
```
Comment (azime) [inline source/kraken/tests/fill_disruption_from_chaos_tests.cpp]: minor: Long line

**Review by TeXitoi — APPROVED** [review]

File: source/kraken/make_disruption_from_chaos.cpp [hunk 2]
```diff
@@ -36,63 +36,73 @@ namespace bt = boost::posix_time;
...
+        update_tag(*tag, chaos_tag);
+        return tag;
+    }
 
     auto tag = boost::make_shared<nt::disruption::Tag>();
-    tag->uri = chaos_tag.id();
-    tag->name = chaos_tag.name();
-    tag->created_at = from_posix(chaos_tag.created_at());
-    tag->updated_at = from_posix(chaos_tag.updated_at());
+    update_tag(*tag, chaos_tag);
```
Comment (xlqian) [inline source/kraken/make_disruption_from_chaos.cpp:57]: how about: 
```c++
auto tag = weak_tag.lock();
tag = tag ? tag : boost::make_shared<nt::disruption::Tag>();
update_tag(*tag, chaos_tag);
```

File: source/kraken/make_disruption_from_chaos.cpp [hunk 3]
```diff
@@ -36,63 +36,73 @@ namespace bt = boost::posix_time;
...
 
     auto cause = boost::make_shared<nt::disruption::Cause>();
-    cause->uri = chaos_cause.id();
-    cause->wording = chaos_cause.wording();
-    cause->created_at = from_posix(chaos_cause.created_at());
-    cause->updated_at = from_posix(chaos_cause.updated_at());
-    if(chaos_cause.has_category()){
-        cause->category = chaos_cause.category().name();
-    }
+    update_cause(*cause, chaos_cause);
```
Comment (xlqian) [inline source/kraken/make_disruption_from_chaos.cpp]:
```c++
auto cause = weak_cause.lock();
cause = cause ? cause: boost::make_shared<nt::disruption::Cause>();
update_cause(*cause, chaos_cause);
```

File: source/kraken/make_disruption_from_chaos.cpp [hunk 4]
```diff
@@ -106,6 +116,18 @@ make_severity(const chaos::Severity& chaos_severity, nt::disruption::DisruptionH
...
+static boost::shared_ptr<nt::disruption::Severity>
+make_severity(const chaos::Severity& chaos_severity, nt::disruption::DisruptionHolder& holder) {
+    auto& weak_severity = holder.severities[chaos_severity.id()];
+    if (auto severity = weak_severity.lock()) {
+        update_severity(*severity, chaos_severity);
+        return severity;
+    }
+
+    auto severity = boost::make_shared<nt::disruption::Severity>();
+    update_severity(*severity, chaos_severity);
```
Comment (xlqian) [inline source/kraken/make_disruption_from_chaos.cpp]: same here

**Review by xlqian — APPROVED** [review]

**Review by TeXitoi — APPROVED** [review]

**Comment by antoine-de** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
Might be merged before release?

**Comment by antoine-de** [comment]
yes I think so, and since it's a bug correction for Chaos I think it would
be nice

File: source/kraken/make_disruption_from_chaos.cpp [hunk 5]
```diff
@@ -36,63 +36,68 @@ namespace bt = boost::posix_time;
...
-    weak_cause = cause;
+    update_cause(*cause, chaos_cause);
     return cause;
 
 }
 
-static boost::shared_ptr<nt::disruption::Severity>
-make_severity(const chaos::Severity& chaos_severity, nt::disruption::DisruptionHolder& holder) {
+static void update_severity(nt::disruption::Severity& severity, const chaos::Severity& chaos_severity) {
     namespace tr = transit_realtime;
```
Comment (pbougue) [inline source/kraken/make_disruption_from_chaos.cpp:88]: minor: can be (re)moved

**Review by pbougue — APPROVED** [review]

**Pull request merged**